src/oom/oomd-manager.c

```diff
@@ -653,6 +668,9 @@ Manager* manager_free(Manager *m) {
         hashmap_free(m->monitored_mem_pressure_cgroup_contexts);
         hashmap_free(m->monitored_mem_pressure_cgroup_contexts_candidates);
 
+        clean_prekills(m->prekill_ctxs);
+        set_free(m->prekill_ctxs);
```

Comment:
If you use proper hash ops for the set then you won't have to call clean_prekills() here, freeing the set will be sufficient.